micro_dl/cli/image_inference.py

```diff
@@ -229,33 +261,64 @@ def run_prediction(args, gpu_ids, gpu_mem_frac):
                     np.save(file_name, im_pred, allow_pickle=True)
                 else:
                     raise ValueError('Unsupported file extension')
+
+                # assuming target and predicted images are always 2D for now
+                # Load target
+                meta_idx = aux_utils.get_meta_idx(
+                    frames_meta,
+                    time_idx,
+                    target_channel,
+                    slice_idx,
+                    pos_idx,
+                )
+                file_path = os.path.join(
+                    args.image_dir,
+                    frames_meta.loc[meta_idx, "file_name"],
+                )
+                # get a single row of frame meta data
+                test_frames_meta_row = frames_meta.loc[[meta_idx]]
+                im_target = preprocess_imstack(
+                    frames_metadata=frames_meta,
+                    input_dir=args.image_dir,
+                    depth=1,
+                    time_idx=time_idx,
+                    channel_idx=target_channel,
+                    slice_idx=slice_idx,
+                    pos_idx=pos_idx,
+                )
+                im_target = image_utils.crop2base(im_target)
+                if data_format == 'channels_first':
+                    # Change dimension order to zyx (3D) or channel first (2D)
+                    im_target = np.transpose(im_target, [2, 0, 1])
+                im_target = im_target[np.newaxis, ...]
```

Comment:
this is the case for data_format == 'channels_first'

Reply:
Yes, and line 293 is to add the batch dimension

Reply:
So I guess:
# TODO: Add image_format option to network config
im_target = np.transpose(im_target, [2, 0, 1])
if data_format == 'channels_first':
    im_target = im_target[np.newaxis, ...]
else:
    im_target = im_target[..., np.newaxis]